auto_submit/lib/action/git_cli_revert_method.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2023 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:auto_submit/action/revert_method.dart';
+import 'package:auto_submit/configuration/repository_configuration.dart';
+import 'package:auto_submit/git/cli_command.dart';
+import 'package:auto_submit/git/git_access_method.dart';
+import 'package:auto_submit/git/git_cli.dart';
+import 'package:auto_submit/git/git_repository_manager.dart';
+import 'package:auto_submit/git/git_revert_branch_name.dart';
+import 'package:auto_submit/requests/exceptions.dart';
+import 'package:auto_submit/service/config.dart';
+import 'package:auto_submit/service/github_service.dart';
+import 'package:auto_submit/service/log.dart';
+import 'package:auto_submit/service/revert_issue_body_formatter.dart';
+import 'package:github/github.dart' as github;
+import 'package:github/github.dart';
+import 'package:retry/retry.dart';
+
+class GitCliRevertMethod implements RevertMethod {
```

Comment:
Do you have an estimate how long this method takes to create a revert?

Reply:
It is very fast. Since we do a sparse checkout (only files at the root are checked out) it takes no time at all. A second or two? I'll take a look to see how long exactly though.

Reply:
Looks like 7 seconds from the time the message is seen until the clean up of the cloned directory happens.
![image](https://github.com/flutter/cocoon/assets/32242716/b453e5f9-080b-4d35-a1d4-18409d55c817)